spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -629,6 +629,38 @@
         end
       end
     end
+
+    def stub_auth
+      allow(controller).to receive(:validate_saml_request).and_return(true)
+      allow(controller).to receive(:verify_authn_context).and_return(true)
+      allow(controller).to receive(:confirm_two_factor_authenticated).and_return(true)
+      allow(controller).to receive(:link_identity_from_session_data).and_return(true)
+    end
+
+    context 'user is redirected to IdV' do
```

Comment:
Yes, blank lines are used to separate the different stages of a test. See https://robots.thoughtbot.com/four-phase-test